File: android-native/app/src/main/java/com/flynnai/app/data/SupabaseClient.kt
package com.flynnai.app.data

import android.content.Context
import com.flynnai.app.core.Environment
import io.github.jan.supabase.SupabaseClient
import io.github.jan.supabase.auth.Auth
import io.github.jan.supabase.createSupabaseClient
import io.github.jan.supabase.postgrest.Postgrest

object SupabaseClient {
    fun create(context: Context): SupabaseClient = createSupabaseClient(
        supabaseUrl = Environment.supabaseUrl,
        supabaseKey = Environment.supabaseAnonKey,
    ) {
        install(Auth)
        install(Postgrest)
    }
}

File: desktop-mac/FlynnDesktop/Networking/SupabaseClient.swift
import Foundation
import Supabase

/// Singleton Supabase client. Uses the same project as the iOS app.
enum FlynnSupabase {
    static let supabaseURL = URL(string: "https://zvfeafmmtfplzpnocyjw.supabase.co")!

    /// Anon key — safe to bundle (RLS enforces data access).
    static let anonKey = ProcessInfo.processInfo.environment["SUPABASE_ANON_KEY"]
        ?? "eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJzdXBhYmFzZSIsInJlZiI6Inp2ZmVhZm1tdGZwbHpwbm9jeWp3Iiwicm9sZSI6ImFub24iLCJpYXQiOjE3NTYyMDE1NDMsImV4cCI6MjA3MTc3NzU0M30.PnSY6rFvczDiDucsyN0nr-luR_Jb6a6O2uAeZxgBiRI"

    static let client: SupabaseClient = {
        SupabaseClient(
            supabaseURL: supabaseURL,
            supabaseKey: anonKey,
            options: SupabaseClientOptions(
                auth: .init(emitLocalSessionAsInitialSession: true)
            )
        )
    }()
}
